tee-worker/omni-executor/aa-contracts/src/core/EntryPointSimulations.sol

```diff
@@ -90,6 +90,43 @@ contract EntryPointSimulations is EntryPoint, IEntryPointSimulations {
             ExecutionResult(opInfo.preOpGas, paid, validationData, paymasterValidationData, targetSuccess, targetResult);
     }
 
+    /// @inheritdoc IEntryPointSimulations
+    function simulateHandleOps(PackedUserOperation[] calldata ops, address payable beneficiary)
+        external
+        nonReentrant
+        returns (ExecutionResult[] memory results)
+    {
+        uint256 opslen = ops.length;
+        results = new ExecutionResult[](opslen);
+        UserOpInfo[] memory opInfos = new UserOpInfo[](opslen);
+        
+        unchecked {
+            // For simulation, we need to run validation on each operation first
+            for (uint256 i = 0; i < opslen; i++) {
+                _simulationOnlyValidations(ops[i]);
+                (uint256 validationData, uint256 paymasterValidationData) = _validatePrepayment(i, ops[i], opInfos[i]);
+                
+                // Execute the user operation
+                uint256 paid = _executeUserOp(i, ops[i], opInfos[i]);
+                
+                // Create execution result for this operation
+                results[i] = ExecutionResult(
+                    opInfos[i].preOpGas,
+                    paid,
+                    validationData,
+                    paymasterValidationData,
+                    true, // targetSuccess - no target call in batch simulation
+                    "" // targetResult - no target call in batch simulation
+                );
+            }
+        }
+        
+        // Note: In simulation, we don't actually compensate the beneficiary
+        // This is just for simulation purposes to understand gas costs and execution flow
```

Comment:
The compensation also consumes gas :) (tho very little) - shall we add it for completeness?

Reply:
done 👍🏼